### ISSUE TITLE
A new option for `prefer-lookaround`

### ISSUE THREAD

**Issue by bhsd-harry** [ota-meshi/eslint-plugin-regexp#498]
**Motivation**
Lookbehind assertions are not yet implemented by Safari, and they cannot be transpiled by commonly used transpilers either. Therefore, it will be great to have an option such as `lookaheadOnly`.

**Description**
When setting `"lookaheadOnly"` (or `{lookaheadOnly: true}`), the rule will neither report nor fix regular expressions that can be converted to lookbehind assertions.

**Examples**

I apologize if the examples below are invalid, but I wish my proposal is clear to you.

```js
// eslint-enable regexp/prefer-lookaround: ["error", "lookaheadOnly"]
/* ✓ GOOD */
"ab".replace(/(a)b/, "$1c");

/* ✗ BAD */
"ab".replace(/a(b)/, "c$1");
```

**Comment by RunDevelopment** [comment]
WebKit recently (2 weeks ago) merged a lookbehind PR https://github.com/WebKit/WebKit/pull/7109 and fixed their [5-year-old lookbehind issue](https://bugs.webkit.org/show_bug.cgi?id=174931). With a bit of luck, Safari will support lookbehinds in a few months. The release note for [16.3 beta](https://developer.apple.com/documentation/safari-release-notes/safari-16_3-release-notes) don't mention this feature, so it might get released in 16.4.

That being said, it's still worth it to add a `lookaheadOnly: boolean` or `lookbehind: boolean` option to support current/older versions of Safari.

**Comment by bhsd-harry** [comment]
> WebKit recently (2 weeks ago) merged a lookbehind PR [WebKit/WebKit#7109](https://github.com/WebKit/WebKit/pull/7109) and fixed their [5-year-old lookbehind issue](https://bugs.webkit.org/show_bug.cgi?id=174931). With a bit of luck, Safari will support lookbehinds in a few months.

Glad to know that, and thanks a lot for your PR!
